### PR TITLE
Rework K2 stream API

### PR DESCRIPTION
**Pull request by apolyakov**
This PR introduces updated stream API for component communication. It keeps high-level API only:
ComponentQuery class.

### CODE REVIEW

File: runtime-light/stdlib/component/component-api.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+#include "runtime-light/utils/context.h"
+
+// === ComponentQuery =============================================================================
+
+struct C$ComponentQuery final : public refcountable_php_classes<C$ComponentQuery> {
+  uint64_t stream_d{INVALID_PLATFORM_DESCRIPTOR};
+
+  explicit constexpr C$ComponentQuery(uint64_t stream_d_) noexcept
+    : stream_d(stream_d_) {}
+  constexpr C$ComponentQuery(const C$ComponentQuery &) noexcept = default;
```
Comment (astrophysik) [inline runtime-light/stdlib/component/component-api.h:24]: Why `C$ComponentQuery` has copy constructor?

File: runtime-light/stdlib/exit/exit.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+task_t<void> shutdown_script() noexcept {
+  auto &component_ctx{*get_component_context()};
+  const auto standard_stream{component_ctx.standard_stream()};
+  if (standard_stream == INVALID_PLATFORM_DESCRIPTOR) {
+    component_ctx.poll_status = PollStatus::PollFinishedError;
+    co_return;
+  }
+
+  const auto ob_total_buffer{ob_merge_buffers()};
```
Comment (astrophysik) [inline runtime-light/stdlib/exit/exit.cpp:41]: Is it good idea to set `auto` everywhere? As for me it makes the code harder to read

File: runtime-light/stdlib/string/string-api.h [hunk 3]
```diff
@@ -0,0 +1,11 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2024 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#include <cstdint>
+
+#include "runtime-core/runtime-core.h"
+
+inline int64_t f$strlen(const string &s) {
```
Comment (astrophysik) [inline runtime-light/stdlib/string/string-api.h:9]: Why this file isn't derived from string-functions?

Comment (astrophysik) [inline runtime-light/stdlib/string/string-api.h:9]: noexcept?)

File: builtin-functions/kphp-light/functions.txt [hunk 4]
```diff
@@ -215,11 +195,6 @@ function warning($message ::: string) ::: void;
 /** @kphp-no-return */
 function critical_error($message ::: string) ::: void;
 
-function debug_print_string($str ::: string) ::: void;
-
-function byte_to_int($str ::: string) ::: ?int;
```
Comment (astrophysik) [inline builtin-functions/kphp-light/functions.txt:220]: I added this functions for proptest. Maybe, we really doesn't need them anymore

Comment (apolyakov) [reply builtin-functions/kphp-light/functions.txt:220]: I think that we need to rewrite proptest with new component API, so we really don't need these functions now.

File: runtime-light/stdlib/component/component-api.cpp [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+#include "runtime-light/component/component.h"
+#include "runtime-light/coroutine/awaitable.h"
+#include "runtime-light/coroutine/task.h"
+#include "runtime-light/streams/streams.h"
+#include "runtime-light/utils/context.h"
+
+// === component query client interface ===========================================================
+
+task_t<class_instance<C$ComponentQuery>> f$component_client_send_request(string name, string message) noexcept {
+  const auto stream_d{get_component_context()->open_stream(name)};
```
[the remark this answers is not in the document]
Comment (apolyakov) [reply runtime-light/stdlib/component/component-api.cpp:20]: Fixed

File: runtime-light/stdlib/output/print-functions.h [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+inline void print(const string &s) noexcept {
+  print(s.c_str(), s.size());
+}
+
+inline int64_t f$print(const string &s) noexcept {
+  print(s);
+  return 1;
+}
+
+// === print_r ====================================================================================
```
Comment (astrophysik) [inline runtime-light/stdlib/output/print-functions.h:37]: As for me, too much comments. But okay

**Review by astrophysik — DISMISSED** [review]

**Review by astrophysik — APPROVED** [review]

**Pull request merged**